### ISSUE TITLE
add enablePrivateServiceConnect to google_vertex_ai_endpoint

### ISSUE THREAD

**Issue by imrannayer** [hashicorp/terraform-provider-google#17836]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team) to expedite investigation and resolution of this issue.

### Description

add enablePrivateServiceConnect to google_vertex_ai_endpoint

### New or Affected Resource(s)

* google_vertex_ai_endpoint

### Potential Terraform Configuration

```tf 
resource "google_vertex_ai_endpoint" "endpoint" {
  name         = "endpoint-name"
  display_name = "sample-endpoint"
  description  = "A sample vertex endpoint"
  location     = "us-central1"
  region       = "us-central1"

  enable_Private_Service_Connect = true

  depends_on   = [
    google_service_networking_connection.vertex_vpc_connection
  ]
}
```

### References

https://cloud.google.com/vertex-ai/docs/general/psc-endpoints

b/334930807

**Comment by SarahFrench** [comment]
Note prior to the triage later today: the [enablePrivateServiceConnect field appears to be deprecated](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.endpoints#Endpoint.FIELDS.enable_private_service_connect) in the API.

**Comment by imrannayer** [comment]
I am not sure why it is shown deprecated when this is a prefered way of exposing endpoint.

**Comment by SarahFrench** [comment]
Note from triage: Adding support for deprecated fields isn't usually a blocker for adding provider support, so we're forwarding to the relevant service team. If there's a recommended alternative they'll know

**Comment by rm-jeremyduplessis** [comment]
Has there been any progress on this?

**Comment by imrannayer** [comment]
fixed in [6.11](https://registry.terraform.io/providers/hashicorp/google/6.11.0/docs/resources/vertex_ai_endpoint#private_service_connect_config-4)
